test/common/ssl/context_impl_test.cc

```diff
@@ -37,6 +39,59 @@ TEST_F(SslContextImplTest, TestdNSNameMatching) {
   EXPECT_FALSE(ContextImpl::dNSNameMatch("lyft.com", ""));
 }
 
+TEST_F(SslContextImplTest, TestGetSubjectAlternateNamesWithDNS) {
+  FILE* fp = fopen(
+      TestEnvironment::runfilesPath("test/common/ssl/test_data/san_dns_cert.pem").c_str(), "r");
+  EXPECT_NE(fp, nullptr);
+  X509* cert = PEM_read_X509(fp, nullptr, nullptr, nullptr);
+  EXPECT_NE(cert, nullptr);
+  const std::vector<std::string>& subject_alt_names = Utility::getSubjectAltNames(*cert, GEN_DNS);
+  EXPECT_EQ(1, subject_alt_names.size());
+  X509_free(cert);
+  fclose(fp);
+}
+
+TEST_F(SslContextImplTest, TestMultipleGetSubjectAlternateNamesWithDNS) {
+  FILE* fp = fopen(
+      TestEnvironment::runfilesPath("test/common/ssl/test_data/san_multiple_dns_cert.pem").c_str(),
+      "r");
+  EXPECT_NE(fp, nullptr);
+  X509* cert = PEM_read_X509(fp, nullptr, nullptr, nullptr);
+  EXPECT_NE(cert, nullptr);
+  const std::vector<std::string>& subject_alt_names = Utility::getSubjectAltNames(*cert, GEN_DNS);
+  EXPECT_EQ(2, subject_alt_names.size());
+  X509_free(cert);
+  fclose(fp);
+}
+
+TEST_F(SslContextImplTest, TestGetSubjectAlternateNamesWithUri) {
+  FILE* fp = fopen(
```

Comment:
use TestEnvironment::readFileToStringForTest read into a string then use BIO? See how context_impl reads certificate. Also factor out that function for these 3 tests.

Reply:
ok. I will refactor to this to a function in utility test using the same approach because rest of the context_test_impl also uses the same to be consistent.

Reply:
Ah ok, filed #4635 to track them.